### PR TITLE
docs: update examples

### PR DESCRIPTION
**Pull request by Gavin-Niederman**
## Describe the changes this PR makes. Why should it be merged?

This PR updates the examples in the `vexide` crate to work with the new APIs.
It also adds a few new examples and adds a CI job to make sure that examples don't become out of date in the future.

### CODE REVIEW

File: packages/vexide-startup/examples/booting.rs [hunk 1]
```diff
@@ -7,10 +7,9 @@ use core::time::Duration;
 
 use vexide_async::sleep;
 use vexide_core::println;
-use vexide_devices::peripherals::Peripherals;
```
Comment (Tropix126) [inline packages/vexide-startup/examples/booting.rs:10]: Is there any real reason for this example to stay around (at least here)?

File: packages/vexide/examples/competition.rs [hunk 2]
```diff
@@ -8,16 +8,16 @@ use alloc::boxed::Box;
 
 use vexide::prelude::*;
 
-#[vexide_startup::main]
```
Comment (Tropix126) [inline packages/vexide/examples/competition.rs:11]: I think this example should be renamed to `competition_builder` or repurposed to just use `CompetitionRobot` to not suggest to people that this is the recommended way to handle the competition api.

File: packages/vexide/examples/smart.rs [hunk 3]
```diff
@@ -0,0 +1,53 @@
+#![no_std]
+#![no_main]
+
+use core::time::Duration;
+
+use vexide::prelude::*;
```
Comment (Tropix126) [inline packages/vexide/examples/smart.rs:6]: nitpick: might consider renaming this example to `smart_devices` just to be more obvious at first glance

**Review by Tropix126 — CHANGES_REQUESTED** [review]

File: packages/vexide-startup/examples/booting.rs (hunk 1 above)
Comment (Gavin-Niederman) [reply packages/vexide-startup/examples/booting.rs:10]: I have updated the example so that it's actually useful instead of removing it.

**Review by Tropix126 — APPROVED** [review]
LGTM

**Pull request merged**